Source/WebInspectorUI/UserInterface/Base/BrowserInspectorFrontendHost.js

```diff
@@ -226,13 +226,25 @@ if (!window.InspectorFrontendHost) {
 
         canSave(saveMode)
         {
-            return false;
+            // Support single file mode only.
+            return saveMode === WI.FileUtilities.SaveMode.SingleFile;
         }
 
         save(saveDatas, forceSaveAs)
         {
-            // FIXME: Create a Blob from the content, get an object URL, open it to trigger a download.
-            throw "unimplemented";
+            console.assert(saveDatas.length === 1, saveDatas);
+
+            let saveData = saveDatas[0];
+            let blob = WI.BlobUtilities.blobForContent(saveData.content, saveData.base64Encoded, "octet/stream");
+            let blobURL = window.URL.createObjectURL(blob);
+
+            let a = document.createElement("a");
+            a.href = blobURL;
+            a.style.display = "none";
+            // url looks like "web-inspector:///file.ext" use filename only
+            a.download = saveData.url.split("/").pop();
+            a.click();
```

Comment:
will this work if `InspectorFrontendHost.save` was not called in response to user activation?

Reply:
I think it will work. I didn't test calling InspectorFrontendHost.save, but I've created a custom test that uses that very same approach to download a file from a setTimeout call, without user interaction, and it works as expected.